Artesian/Artesian.SDK/Service/Clients/Client.cs

```diff
@@ -147,21 +148,59 @@ public Client(IArtesianServiceConfig config, string Url, ArtesianPolicyConfig po
                         {
                             var responseText = await res.Content.ReadAsStringAsync();
                             var exceptionMessage = $"Failed handling REST call to WebInterface {method} {_url + resource}. Returned status: {res.StatusCode}. Content: \n";
+                            var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
 
                             if (res.StatusCode == HttpStatusCode.BadRequest)
                             {
-                                var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
+                                if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
+                                {
+                                    ArtesianSdkProblemDetail problemDetail = JsonConvert.DeserializeObject<ArtesianSdkProblemDetail>(responseDeserialized.ToString());
+
+                                    if (problemDetail.Detail != null)
+                                    {
+                                        responseText = problemDetail.Detail;
+                                    }
+                                    else if (problemDetail.Extensions["Errors"].Count() > 0)
+                                    {
+                                        responseText = "";
+                                        for (int i = 0; i < problemDetail.Extensions["Errors"].Count(); i++)
+                                        {
+                                            responseText += problemDetail.Extensions["Errors"][i]["Value"][0]["ErrorMessage"];
+                                        }
+                                    }
+                                    else if (problemDetail.Title != null)
+                                    {
+                                        responseText = problemDetail.Title;
+                                    }
+                                    throw new ArtesianSdkValidationException(exceptionMessage + responseText, problemDetail);
+                                }
+
                                 responseText = _tryDecodeText(responseDeserialized);
 
                                 throw new ArtesianSdkValidationException(exceptionMessage + responseText);
                             }
 
+                            if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
```

Comment:
I this this can be consolidated reorganizing the code

```
ArtesianSdkProblemDetail pd = null;
string rawContent = string.Empty;

if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
{
   pd = await res.Content.ReadAsAsync<ArtesianSdkProblemDetail>(_formatters, ctk);
} else {
   rawContent = await res.Content.ReadAsStringAsync();
}

var detailMessage = pd?.Details ?? pd?.Title ?? pd?.Type ?? "Content:" + Environment.NewLine + rawContent;
var exceptionMessage = $"Failed handling REST call to WebInterface {method} {_url + resource}. Returned status: {res.StatusCode}. {detailMessage}";

switch (res.StatusCode)
{
    case HttpStatusCode.BadRequest:
         throw new ArtesianSdkValidationException(exceptionMessage, pd);
    case HttpStatusCode.Conflict:
    case HttpStatusCode.PreconditionFailed:
         throw new ArtesianSdkOptimisticConcurrencyException(exceptionMessage, pd);
    case HttpStatusCode.Forbidden:
         throw new ArtesianSdkForbiddenException(exceptionMessage, pd);
    default:
         throw new ArtesianSdkRemoteException(exceptionMessage, pd);
}
```

Reply:
_tryDecodeText case for previous 400 case. Is this the ideal way to handle this?

if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
                            {
                                problemDetail = await res.Content.ReadAsAsync<ArtesianSdkProblemDetail>(_formatters, ctk);
                            }
                            else
                            {
                                if (res.StatusCode == HttpStatusCode.BadRequest)
                                {
                                    responseText = _tryDecodeText(await res.Content.ReadAsAsync<ArtesianSdkProblemDetail>(_formatters, ctk));
                                }
                                else
                                {
                                    responseText = await res.Content.ReadAsStringAsync();
                                }
                            }

Reply:
flow is ok, but you're missing <T> on the ReadAsync calls.